Saturation/SaturationAlgorithm.cpp

```diff
@@ -1308,14 +1307,12 @@ UnitList* SaturationAlgorithm::collectSaturatedSet()
 {
   CALL("SaturationAlgorithm::collectSaturatedSet");
 
-  LiteralIndexingStructure* gis=getIndexManager()->getGeneratingLiteralIndexingStructure();
-
   UnitList* res = 0;
-  SLQueryResultIterator qrit = gis->getAll();
-  while (qrit.hasNext()) {
-    SLQueryResult qres = qrit.next();
-    UnitList::push(qres.clause, res);
-    qres.clause->incRefCnt();
+  ClauseIterator it = _active->clauses();
+  while (it.hasNext()) {
```

Comment:
We're iterating over a set. This makes me worried about non-determinism. If we have two proof searches that end in the same saturated set do these get printed in exactly the same way? If not that's a potential sub-optimal user experience.

Reply:
Yes, this won't be guaranteed. We need to hash the _content_ of a `Clause` somehow, probably via `DerefPtrHash`.

Reply:
Well spotted! I think we could easily hash on `clause->number()` as these are as unique as the pointers themselves. @MichaelRawson , you are the primary/secondary hash master now, what should I override/overload and where, to get the clauses' hash here be derived from `number()`?